### ISSUE TITLE
Wishlist: Weblink in signature of "clean" emails to allow reporting to spam (sa-learn)

### ISSUE THREAD

**Issue by quintingsnj** [mailwatch/MailWatch#936]
EFA has this capability - I'm assuming it's something that the guy(s) that put the EFA VM together have written, but I'm wondering if something like that can be incorporated into MW.

Thanks.

**Comment by asuweb** [comment]
How does this currently work with EFA?  I'm assuming this would require that clean emails are stored on the server to allow for sa-learn when clicking the weblink in the email.

I have plans to implement a web upload feature for sa-learn, which will in part assist with this.

**Comment by quintingsnj** [comment]
Yes, this requires that clean emails are stored on the server. Looking at the code, it passes the mail ID and a token (I think the token is a field in the table so that the link can "expire" after 3 days)...and then a call to sa-learn with the right message ID, and a couple of web pages to report success or error or expired token.

**Comment by asuweb** [comment]
Thanks - that's what I assumed. :)

**Comment by Gatorfreak** [comment]
Can I add a big "Yes, please!" to this suggestion?  :-)

**Comment by Skywalker-11** [comment]
related to #854
